tony-core/src/test/java/com/linkedin/tony/TestTonyE2E.java

```diff
@@ -222,4 +224,27 @@ public void testAMStopsJobAfterTimeout() {
     }, conf);
     Assert.assertEquals(exitCode, -1);
   }
+
+  /**
+   * Test that makes sure if a worker is killed due to OOM, AM should stop the training (or retry).
+   * This test might hang if there is a regression in handling the OOM scenario.
+   *
+   * The reason why we use a Constants.TEST_WORKER_TERMINATED flag instead of simply requesting more memory than
+   * allocated is that Physical Memory Enforcement doesn't seem to work under MiniYARN.
+   */
+  @Test
+  public void testAMStopsJobAfterWorkerKilledByOOM() {
```

Comment:
I think you should rename this test `testAMStopsJobAfterWorker0Killed` (remove the `ByOOM` part) because the test doesn't actually simulate an OOM -- it just simulates worker 0 getting killed, which could be due to any number of reasons.

Reply:
Yah, will update this. I was planning to simulate OOM at the beginning but PMEM enforcement doesn't seem to work under MiniYARN cluster.